Add Development config files

kauberry opened the pull request:
### Description

Adding in config files for the docker image version of a basic Symfony app, complete with Nginx config

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

kauberry-pnnl committed:
Updating symfony main version to 3.3

kauberry-pnnl committed:
Committing Dockerfile and compose configuration elements, along with composer files

kauberry-pnnl committed:
Trimming out useless bits from compose file

kauberry-pnnl committed:
Updated README file and Nginx application config

Commit:
Created bare-bones Symfony application and ported Pacifica Search demonstration code into it

kauberry-pnnl committed:
Fixing issues with weird ElasticSearch configuration

Commit:
Added and configured FOSRestBundle with a skeleton of a filter/result interface

Commit:
Removed all of the old JS client and replaced it with a new REST interface with most of the functionality mocked up

Commit:
Enable/disable filter options based on what is returned by the REST API

Commit:
Refactored all DOM selectors into a DOM manager class and moved attribute retrieval into a utility assertion method

Commit:
Simple debugging - fixing errors caused by previous commit

Commit:
Created basic structure for a query-builder based system with repositories and model classes. Fully implemented InstrumentType.

Commit:
Changed getter to generic getAll(), this will allow us to generically treat repositories in the SearchController as we add them

Commit:
Added the remaining model and repository types - the UI now displays all options for all filter categories

Commit:
Rename variables

Commit:
Sort displayed values by name

Commit:
Removed unused use statement

Commit:
Fixed preg_replace regexes - don't treat capital letter at the beginning of a string as a separation between words

Commit:
Switch filter keys to singular to be consistent with server's naming conventions

Commit:
Added JMS Serializer Bundle to composer.json, and added required parameters to the .dist file

Commit:
Added error handling for the event that a type returns an empty set of records

Commit:
Moved public getAll() method above protected/private methods

Commit:
Added Transaction relationships to all models and restricted the UI to only show filter options associated with transactions

Commit:
Fixed a bug that caused all filter options to be submitted to the server regardless of what was checked

Commit:
Quality of life improvement for development environment - let a dummy host be specified that is automatically replaced with the correct IP address

Commit:
Fixed handling of filter search response and added a label class to make it easier to see which items are enabled/disabled

Commit:
Added XDEBUG to Docker configuration

Commit:
Added the ability to retrieve valid filter options based on current filter values

Commit:
Added an explanatory comment

Commit:
Added support for responses that don't contain all filter types

Commit:
Added casting to ensure that getIdsByTransactionIds() returns values of a consistent type

Commit:
Fix an error that was causing fields not to re-enable after a filter update

Commit:
Trigger update on every change instead of requiring a button push

Commit:
Increased size to a very large value as an interim solution to the problem that we don't yet have handling to scan&scroll over very large result sets.

Commit:
Added an exception to detect when we have a result set that's larger than we can handle.

Commit:
Oops - size was set to a value that's larger than allowed by Elasticsearch

Commit:
Added support for retrieving files on filter update

Commit:
Fixed inconsistent data type being returned by the ProposalRepository

Commit:
Remove unused index.php file

Commit:
Removed unused composer.json file

Commit:
Merge remote-tracking branch 'pacifica/master'

Commit:
Revert "Removed unused composer.json file"

This reverts commit 6620b84e51e9c9dfac062153247395ab3f5f068b.

Commit:
Restore portions of composer.json required to support Travis tests

Commit:
Run composer for both Symfony and global dependencies in install.sh

Commit:
Added .codeclimate.yml to exclude Symfony-generated files from code analysis

Commit:
Updated static analysis's syntax check to exclude the non-existent index.php and to include the source files of our Symfony bundle

Commit:
Removed non-existent index.php from unit-tests.sh output

Commit:
Replaced index.php with Pacifica bundle path in PHP Code Sniffer static analysis

Commit:
Added @return doc comments to controller actions

Commit:
Layout improvements

Commit:
Made TRUE/FALSE/NULL all-caps to align with Pacifia code style guidelines

Commit:
Switch to PSR2 standard for static code analysis

Commit:
Autoformatting to PSR2 standard

Commit:
Commented out dummy test

Commit:
Refactored large blocks of code into their own functions

Commit:
Refactored TransactionRepository::getIdsOfTypeAssociatedWithAtLeastOneTransaction() to have a lower cognitive complexity rating

Commit:
Refactored a for loop into a separate function to reduce the calling function's cognitive complexity

Commit:
Refactored getImplementingClassNames() to reduce cognitive complexity

Commit:
Deleted an extraneous character

Commit:
Refactored getIdsByFilter() to move lower level-of-abstraction logic into its own functions

Commit:
Removed TODO: I've been informed that the real data *will* have orphaned records, and in any case pulling a database dump to populate the UI is unworkable because of the size of the real database

Commit:
Fixed typo

BenjaminStuermer commented:
This PR resolves issue https://github.com/pacifica/pacifica-search/issues/6 by removing index.php

Commit:
Renamed file UnitTest.php -> WebTest.php

Commit:
Removed an action that was used for development testing

kauberry committed:
Adding in a bunch of common third-party scripts, css, images, icons, etc.

kauberry committed:
Docker file additions

Commit:
Created a first set of functional and unit tests, and updated project code as required to accommodate them

kauberry committed:
Adding in some nginx smarts

Commit:
SearchController doesn't use any behavior inherited from Controller, so remove the extends statement

Commit:
Marked the RESTController as untestable for unit testing

Commit:
Fixed controller test's namespace

Commit:
Added unit tests for ElasticSearchType base classes

Commit:
Added uni tests for model classes

Commit:
Added documentation to parameters.yml.dist for how to connect out to the remote snapshot server

kauberry-pnnl committed:
Merge branch 'master' into Theming-Additions

* master:
  Added documentation to parameters.yml.dist for how to connect out to the remote snapshot server
  Added uni tests for model classes
  Added unit tests for ElasticSearchType base classes
  Fixed controller test's namespace
  Marked the RESTController as untestable for unit testing
  SearchController doesn't use any behavior inherited from Controller, so remove the extends statement
  Created a first set of functional and unit tests, and updated project code as required to accommodate them

# Conflicts:
#	application/src/PacificaSearchBundle/Controller/SearchController.php

Commit:
Removed a docker-compose.yml parameter that breaks the Docker container in a new version of Docker

Commit:
Added a Selenium package to allow running Selenium functional tests in the host machine

Commit:
Added files that are used by Dave Brown's Selenium coverage report functionality

Commit:
Fixed errors introduced by updates to the ElasticSearch data structure

Commit:
Cosmetic: Removed unused "use" statement

Commit:
Renamed SearchController -> GuiController to better reflect its role

kauberry-pnnl committed:
Merge commit '134cdf1862d3f93f41e2a2c585a18f043c98478f' into Theming-Additions

* commit '134cdf1862d3f93f41e2a2c585a18f043c98478f':
  Renamed SearchController -> GuiController to better reflect its role
  Cosmetic: Removed unused "use" statement
  Fixed errors introduced by updates to the ElasticSearch data structure
  Added files that are used by Dave Brown's Selenium coverage report functionality
  Added a Selenium package to allow running Selenium functional tests in the host machine
  Removed a docker-compose.yml parameter that breaks the Docker container in a new version of Docker

# Conflicts:
#	application/src/PacificaSearchBundle/Repository/TransactionRepository.php
#	application/src/PacificaSearchBundle/Service/ElasticSearchQueryBuilder.php

Commit:
Made error from failed search requests more informative

Commit:
Added basic support for paging to the API and GUI and used it to replace the "grab everything with a transaction" approach that was used in the prototype

Commit:
Updated the names used for Instruments and Proposals

Commit:
Added a new route to supply file tree values

kauberry-pnnl committed:
Merge commit '1eb692ba6c61f91057adbe0c0b8a4ed60b182aae' into Theming-Additions

* commit '1eb692ba6c61f91057adbe0c0b8a4ed60b182aae':
  Updated the names used for Instruments and Proposals
  Added basic support for paging to the API and GUI and used it to replace the "grab everything with a transaction" approach that was used in the prototype
  Made error from failed search requests more informative

# Conflicts:
#	application/src/PacificaSearchBundle/Resources/views/search.html.twig

kauberry-pnnl committed:
Merge commit 'e6bf7804f005b83b0b26dd785cfd01517fc51a75' into Theming-Additions

* commit 'e6bf7804f005b83b0b26dd785cfd01517fc51a75':
  Added a new route to supply file tree values

# Conflicts:
#	application/src/PacificaSearchBundle/Controller/RestController.php
#	application/src/PacificaSearchBundle/Repository/Repository.php
#	application/src/PacificaSearchBundle/Resources/config/routing.yml

Commit:
Added a route for the lazy loading of files associated with a transaction

Commit:
Added a dummy value for file IDs

Commit:
Added the file ID to the response, make the order of keys consistent with specification (and also easier to read) and fixed a bug that sometimes created an empty folder at the root level.

kauberry-pnnl committed:
Merge commit '445fce04354ed3e6b66faddb6c18447fcf706a50' into Theming-Additions

* commit '445fce04354ed3e6b66faddb6c18447fcf706a50':
  Added the file ID to the response, make the order of keys consistent with specification (and also easier to read) and fixed a bug that sometimes created an empty folder at the root level.
  Added a dummy value for file IDs
  Added a route for the lazy loading of files associated with a transaction

Commit:
Added missing getFilteredPage() method

kauberry-pnnl committed:
Merge commit 'faf7812e4cb1440d15081a00e6f7622417e228cf' into Theming-Additions

* commit 'faf7812e4cb1440d15081a00e6f7622417e228cf':
  Added missing getFilteredPage() method

kauberry-pnnl committed:
Changes to deal with dynamic loading

kauberry-pnnl committed:
Removed an errant debug statement

Commit:
Removed doc comment that was no longer accurate

Commit:
Instead of disabling/enabling options in the filters, load new filter pages when the filter changes

Commit:
Created a new getIdsByType method

Commit:
When a filter option is selected, add it to a separate "selected options" section of the sidebar

Commit:
Made a method protected that's only used by the class itself

Commit:
Restore a conditional statement that was removed in a previous commit but is actually necessary

kauberry-pnnl committed:
Merge commit '15d191aa4c15b5486c56624a773526c80695f3a9' into Theming-Additions

* commit '15d191aa4c15b5486c56624a773526c80695f3a9':
  Instead of disabling/enabling options in the filters, load new filter pages when the filter changes
  Removed doc comment that was no longer accurate

# Conflicts:
#	application/web/js/pacifica_search.js

Commit:
Cosmetic: integer -> int -- just to make my IDE stop warning me that my types are inconsistent

Commit:
Exclude the IDs of currently filtered items from the results of the filter

kauberry-pnnl committed:
Merge commit 'b47b95b2e3e05a4231ce92eee91f77552551d8d9' into Theming-Additions

* commit 'b47b95b2e3e05a4231ce92eee91f77552551d8d9':
  Exclude the IDs of currently filtered items from the results of the filter
  Cosmetic: integer -> int -- just to make my IDE stop warning me that my types are inconsistent
  Restore a conditional statement that was removed in a previous commit but is actually necessary
  Made a method protected that's only used by the class itself
  When a filter option is selected, add it to a separate "selected options" section of the sidebar
  Created a new getIdsByType method

# Conflicts:
#	application/src/PacificaSearchBundle/Resources/views/search.html.twig

Commit:
Added a hack to make the filter behave as if the database contained only values associated with transactions

Commit:
Added a count() method - it's not used at the moment because I backed out the code that called it but I'll probably need it soon

Commit:
Handle removing an option from the filter

kauberry-pnnl committed:
styling and ui tweaks

kauberry-pnnl committed:
Merge commit '4531e2f73331d1853c53cd4f79c099785bf13f9d' into Theming-Additions

* commit '4531e2f73331d1853c53cd4f79c099785bf13f9d':
  Handle removing an option from the filter
  Added a count() method - it's not used at the moment because I backed out the code that called it but I'll probably need it soon
  Added a hack to make the filter behave as if the database contained only values associated with transactions

# Conflicts:
#	application/web/js/pacifica_search.js

Commit:
Fix a bug that stopped the page number from updating after changing

kauberry-pnnl committed:
Fixing transactions page number call

kauberry-pnnl committed:
Minor style fixes courtesy of phpcbf/phpcs

Commit:
Merge remote-tracking branch 'origin/Theming-Additions'

Commit:
Use a terms query instead of an ids query (ids queries don't work in this version of Elastic Search)

Commit:
Added exceptions to get better failure information if queries break again

Commit:
Fixed an incorrect relationship reference

Commit:
Instead of creating a filter with only the Proposals, which has the effect of loading *all* files inside the Proposals in the current filter, retrieve additional records using the original filter so that, for example, if the filter contains an Instrument other instruments are not included in the file list.

Commit:
Moved functionality from ProposalRepository into Repository because it is generic and could be used by any Repository class

Commit:
Changed handling of a missing filter in the FileTree page getter so that it doesn't try to retrieve files if the filter is empty

Commit:
Removed an unused library file

Commit:
Cosmetic: Cleaned up the indentation of search.html.twig

Commit:
Cosmetic: Cleaned up the indentation of search.html.twig

Commit:
Added debouncing to the AJAX call that loads the filter results

Commit:
Fixed an error - the name of the institutions key in the users type changed

Commit:
Limit transactions handled to 1000

Commit:
Limit transactions handled to 1000

Commit:
Added handling for the case that only currently selected options remain as legal filter options, which was causing the server to return all legal options instead of no legal options

Commit:
Quality of life changes that make it easier to see what the new state of the filter is while waiting for it to load

Commit:
Quality of life changes that make it easier to see what the new state of the filter is while waiting for it to load

kauberry-pnnl committed:
Putting loading animation in place

kauberry-pnnl committed:
Merge branch 'master' of github.com:kauberry/pacifica-search

* 'master' of github.com:kauberry/pacifica-search:
  Quality of life changes that make it easier to see what the new state of the filter is while waiting for it to load
  Quality of life changes that make it easier to see what the new state of the filter is while waiting for it to load
  Added handling for the case that only currently selected options remain as legal filter options, which was causing the server to return all legal options instead of no legal options
  Limit transactions handled to 1000
  Limit transactions handled to 1000
  Fixed an error - the name of the institutions key in the users type changed

# Conflicts:
#	application/web/js/pacifica_search.js

kauberry-pnnl committed:
Working on additions to theming and enhancing information content

Commit:
Fix a bug that causes paging to fail on an empty filter

Commit:
Fixed a bug that caused the file tree to load invalid data if the requested page had no proposals on it

Commit:
Change handling of the filter container's visibility such that the border between the filter sections is only shown when both sides contain elements

kauberry-pnnl committed:
Fixed status page link

kauberry-pnnl committed:
Fixed visual bug in the options selectors that left empty divs

kauberry-pnnl committed:
Merge branch 'master' of github.com:kauberry/pacifica-search

* 'master' of github.com:kauberry/pacifica-search:
  Change handling of the filter container's visibility such that the border between the filter sections is only shown when both sides contain elements
  Fixed a bug that caused the file tree to load invalid data if the requested page had no proposals on it

# Conflicts:
#	application/web/css/combined.css
#	application/web/js/pacifica_search.js

Commit:
Removed CSS that was re-added in a previous merge by mistake

Commit:
Added a console.log() call that provides information about the current version

kauberry committed:
Updating Main search view

kauberry committed:
Merge branch 'master' of github.com:kauberry/pacifica-search

* 'master' of github.com:kauberry/pacifica-search:
  Added a console.log() call that provides information about the current version

kauberry committed:
Pushing up some example queries for use in ES with aggregations

kauberry committed:
Adding in queries for creating ES aliases for existing transactions, last month/week transactions

kauberry committed:
Adding some api results "pseudocode" as examples

Commit:
Corrected caption

Commit:
Removed a variable declaration whose variable was never used

Commit:
Just refer to the boolean "checked" property instead of assigning it to a variable first

Commit:
Cleanup: Removed unused and commented-out code, fixed an unsafe comparison to NULL

Commit:
Removed commented-out code

Commit:
Fixed application.conf::server::location::try_files so that query string is not discarded when the request is passed off to app_dev.php

Commit:
Removed redundant "type" attribute

Commit:
Corrected error message

Commit:
Added text search

kauberry committed:
Changes to the dockerfile and compose file that should clear up xdebug issues in atom, plus changes to extend the
timeout period for the application stack

kauberry committed:
Merge branch 'master' of github.com:kauberry/pacifica-search

* 'master' of github.com:kauberry/pacifica-search:
  Added text search

Commit:
XDEBUG setting update

Commit:
Renamed variables to be more informative

Commit:
Added array_unique calls wherever we're passing lists of IDs around

Commit:
Return the promise created by $.get() so that the promise returned by the enclosing $.ajax call won't resolve until after the $.get() call also resolves. This fixes a bug that was causing the loading animation to end prematurely.

Commit:
Update all RESTful interactions to post the filter's contents rather than relying on server-side session storage.

Commit:
Removed unused code

Commit:
Added a missing semicolon

Commit:
Changed a variable name to be more expressive

Commit:
Revert previous change of method for the transactionFilesAction - post is not necessary because the filter is not sent as part of the request, and the payload that is sent is pretty small

Commit:
Refactored the landing page to load empty filters initially

Commit:
Removed the stopgap getIdsOfTypeAssociatedWithAtLeastOneTransaction(), which will no longer be used to populate the initial values of the facets since we're now initiating searching with a text.

Commit:
Changed types to use lower-case names to reflect that change in the Elasticsearch database. Moved valid types from a private method into a constant since PHP 7.1 allows scoped constants

Commit:
We no longer need custom name handling for instruments, since all models have a display_name field

Commit:
Removed a comment referring to an obsolete feature

Commit:
Created a getPage method for transactions to support the text search feature

Commit:
Updated getText_searchAction to search Transactions and then load matching faceted search types

Commit:
Cleanup: Added typehinting and @throws annotations. Removed the "devserver" magic string because we found a better solution using a builtin Mac host DNS name in Docker

Commit:
Removed the $assertResultsFound parameter from getResults() because we weren't using it anywhere any more

Commit:
Added getIdsByTextString() to the TransactionRepository

Commit:
We no longer need custom name handling for users, since all models have a display_name field

Commit:
Center the loading animation

Commit:
Rebuild application to accommodate new data model

Commit:
Fix a bug in the reported number of transactions matching the filter

Commit:
Removed debugging code

Commit:
Removed debugging code

Commit:
Updated pagination support - pagination is now a simple "load more" link rather than discrete pages, and pagination functionality is now compatible with the new model structure

Commit:
Removed unused constant

Commit:
Removed unnecessary white space

Commit:
Merge pull request #1 from kauberry/feature/new_schema

Feature/new schema

Commit:
Added handling for string-style IDs in queries that exclude IDs

Commit:
Added handling for string-style IDs in Filter::getIdsByType()

Commit:
Refactored getTransactionIdsByOwnIds() to use an aggregation

Commit:
Removed TODO - it is done

Commit:
Return all results instead of an empty result from getPageByTransactionIds() if no transactions are passed

Commit:
Updated all ID references to be strings rather than integers

Commit:
Merge pull request #2 from kauberry/feature/new_schema

Feature/new schema

plithnar committed:
File Tree viewer pulling live data from metadata server

plithnar committed:
Update for Proposal display name to be the display text

plithnar committed:
Updating default operator to 'AND' from 'OR' for text searches.
Updating fileTree to be show the same Proposal information as the facets would.

plithnar committed:
Moving files from /web/css and /web/js to /web/assets/css and /web/assets/js in an effort to make the deployed site work better/easier.

plithnar committed:
Update to make facets appear in descending order of transaction count.

plithnar committed:
Adding React, ReactDOM and Searchkit for use with elasticsearch and displaying the results.

plithnar committed:
Searchkit with aggregations now functional

plithnar committed:
Update for instrument_groups to be their own aggregation.

plithnar committed:
Update to pull data from configs

plithnar committed:
Updated composer file due to no longer requiring elasticsearch

plithnar committed:
removal of npm install calls to fix the composer update and composer install calls

plithnar committed:
Update to searchkit to support AND and OR in the search query

plithnar committed:
Addition of Informative tooltip for search box

plithnar committed:
Update to react file to use className instead of class. This is the correct format for react elements.

plithnar committed:
Update to correct metadata requests

plithnar committed:
Beginning of refactor to remove unneeded code

plithnar committed:
Update from dataTooltip to datatooltip for React compatability.

plithnar committed:
Styling updates to make files not appear as links.

makopov commented on application/src/PacificaSearchBundle/Filter.php:
```diff
@@ -0,0 +1,308 @@
+     * @param string[] $ids
+     * @return $this
+     */
+    public function setIdsByType(string $class, array $ids) : Filter
```
Alternatively, you could specify the type "ElasticSearchType" in place of "string", php 7 will throw a type error that the calling function could catch. Eliminating the need for the check below as well as in the next function.

BenjaminStuermer replied:
That doesn't work - the argument is the name of a class, not an instance of the class.

```
class Base {}
class Sue extends Base {}
function doTheThing(Base $thing) {}
doTheThing(Sue::class);
// -> Exception: Argument 1 passed to doTheThing() must be an instance of Base, string given
```

makopov commented on application/tests/PacificaSearchBundle/Model/ElasticSearchTypeCollectionTest.php:
```diff
@@ -0,0 +1,69 @@
+     */
+    public function basicFunctionality()
+    {
+        $letters = 'abcdefghijklmnopqrstuvwxwzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890_ ';
```
You may want to look into the faker factory for symphony. https://github.com/fzaninotto/Faker

makopov commented on application/tests/PacificaSearchBundle/TestCase.php:
```diff
@@ -0,0 +1,23 @@
+<?php
```
Are these tests being ran in CI anywhere?

plithnar committed:
Added filter for if data is released.

plithnar committed:
Adding placeholder element for the expected dateRange for the translation's upload date.

plithnar committed:
Update to show date-range filter for upload date. More date ranges to come

plithnar committed:
Addition of other date range filters and collapsible panels to help keep the facets from getting cluttered.

plithnar committed:
Updated the fontawesome stylesheet to come from https to mesh with deployed project.
Added styling to reduce the amount of whitespace when it is displayed in a high resolution system.

plithnar committed:
Update to provide translation for the date ranges to make it clearer.

plithnar committed:
Update to hide abstracts longer than 3 lines and "link" function to show/hide the full abstract
